### PR TITLE
🔧 refactor nft storage in services

### PR DESCRIPTION
**Pull request by roiLeo**
### PR type

- [x] Refactoring

### What's new?

- [x] PR related with #3379 & #3358
- [x] move nftStorage to `/services`
- [x] replace axios with `$fetch` 

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [ ] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

⚠️ I didn't test if pinning still work with these changes

### CODE REVIEW

File: services/nftStorage.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    throw new Error(
+      `[NFT::STORAGE] Unable to PIN File for reasons ${error.data}`
+    )
+  })
+  consola.log('[NFT::STORAGE] Pin File')
+  return value.cid
+}
+
+export default nftStorageApi
+// QmYt2FydonvVMsEqe2q3hvm38WDq21xM8Z5ZSHZw19PwjF;
```
Comment (kkukelka) [inline services/nftStorage.ts:67]: this shouldn't be publicly available on github right?

File: services/nftStorage.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  })
+  return { expiry, token }
+}
+
+export const pinFileToIPFS = async (file: Blob, _: string): Promise<string> => {
+  const { value } = await nftStorageApi<StorageApiResponse>('/pinFile', {
+    method: 'POST',
+    body: file,
+    headers: {
+      'Content-Type': file.type ? `${file.type}` : '*/*',
```
Comment (kkukelka) [inline services/nftStorage.ts:55]: this should suffice
```suggestion
      'Content-Type': file.type || '*/*',
```

**Review by kkukelka — CHANGES_REQUESTED** [review]

File: services/nftStorage.ts (hunk 1 above)
Comment (roiLeo) [reply services/nftStorage.ts:67]: I don't really know what this key is about maybe, should we hide it? It doesn't seem to be used

Comment (kkukelka) [reply services/nftStorage.ts:67]: yeah let's hide it just to be safe

File: services/nftStorage.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    throw new Error(
+      `[NFT::STORAGE] Unable to PIN File for reasons ${error.data}`
+    )
+  })
+  consola.log('[NFT::STORAGE] Pin File')
+  return value.cid
+}
+
+export default nftStorageApi
+// QmYt2FydonvVMsEqe2q3hvm38WDq21xM8Z5ZSHZw19PwjF;
```
Comment (roiLeo) [inline services/nftStorage.ts:67]:
```suggestion
```

**Review by kkukelka — APPROVED** [review]
gtg now

**Comment by roiLeo** [comment]
Can anyone test if minting works on this branch?
I can't see my bsx balance :(

**Comment by roiLeo** [comment]
> Can anyone test if minting works on this branch? I can't see my bsx balance :(

nvm, it worked:
https://deploy-preview-3741--koda-nuxt.netlify.app/bsx/gallery/2444376227-5

File: services/nftStorage.ts (hunk 1 above)
Comment (yangwao) [reply services/nftStorage.ts]: it's the public hash of the file 

https://nftstorage.link/ipfs/QmYt2FydonvVMsEqe2q3hvm38WDq21xM8Z5ZSHZw19PwjF -> 
https://bafybeie4ty6y2jxljdguxcpnbxxzlujiiycqynrov6zuaqayo4dc3kasti.ipfs.nftstorage.link/

Comment (yangwao) [reply services/nftStorage.ts]: but doesn't need to be there tho 👀

**Comment by DeluneClaire** [comment]
So should I test the whole minting process on BSX and RMRK as well, or?

**Comment by DeluneClaire** [comment]
So should I test the whole minting process on BSX and RMRK as well, or?

**Comment by roiLeo** [comment]
> So should I test the whole minting process on BSX and RMRK as well, or?

well since it worked on bsx I don't think we need to test it again on rmrk

**Comment by petersopko** [comment]
minted rmrk/bsx wfm

**Comment by petersopko** [comment]
@roiLeo deepsource seems to have some issues, can you check?

**Review by DeluneClaire — APPROVED** [review]

**Comment by roiLeo** [comment]
Screenshot 2022-08-18 at 14-10-39 roiLeo chore_services_nftStorage @ 87a7626 • kodadot_nft-gallery

jerry

**Comment by roiLeo** [comment]
Screenshot 2022-08-18 at 14-37-39 roiLeo chore_services_nftStorage @ 9fb3137 • kodadot_nft-gallery

Now deepscan not happy but I don't know why

**Comment by petersopko** [comment]
> Now deepscan not happy but I don't know why

I see the possible issue, but it seems like a pretty random find considering that you haven't even touched that file + that's more of a recommendation (nothing breaks because of it). let's merge 🚀

**Comment by petersopko** [comment]
lol, tricked it! 😆 
image

File: utils/safePin.ts [hunk 4]
```diff
@@ -11,7 +11,7 @@ export async function pinImageSafe(
     return IPFS_KODADOT_IMAGE_PLACEHOLDER
   }
 
-  return pinFileToIPFS(file, token)
+  return await pinFileToIPFS(file, token)
```
Comment (vikiival) [inline utils/safePin.ts:14]: Useless await?

Comment (roiLeo) [reply utils/safePin.ts:14]: since it's async function I don't think so? plus any use of `pinFileToIPFS` in codebase use async/await

**Pull request merged**